pandas/core/arrays/boolean.py

```diff
@@ -686,9 +686,13 @@ def _reduce(self, name: str, skipna: bool = True, **kwargs):
         data = self._data
         mask = self._mask
 
-        if name in {"sum", "min", "max"}:
+        if name in {"sum", "prod", "min", "max"}:
             op = getattr(masked_reductions, name)
-            return op(data, mask, skipna=skipna, **kwargs)
+            result = op(data, mask, skipna=skipna, **kwargs)
+            dtype = maybe_cast_result_dtype(dtype=data.dtype, how=name)
+            if notna(result) and (dtype != result.dtype):
+                result = result.astype(dtype)
```

Comment:
I am still not sure we actually need to do this. We could also choose to follow numpy's behaviour to return platform int (any idea what we do for "bool" dtype?)

BTW, this should also change the result for `sum`, so this was not tested?

Reply:
@jorisvandenbossche I think you're right and it was actually the test that needed to change here (np.int64 -> np.int_)

Reply:
Yeah, in any case that's what I did for sum, I see now (so if we decide on following numpy vs always returning int64, we should do it for both sum and prod)

Reply:
(but I am fine with following numpy)